urlaf656038a0_Salihan04_MOBLIMA_tgz-pJ8-com_moblima_util_dataTestJ8/src/com/moblima/util/dataGenerator.java

```diff
@@ -0,0 +1,600 @@
+		movieDAO.createMovie(movie5);
+		Movie movie6 = new Movie("Marvels Thor The Dark World(3D) ", MovieRating.PG13, MovieType._3D,false, MovieStatus.NOW_SHOWING, "English with Chinese Subtitles", "Chris Hemsworth, Natalie Portman, Anthony Hopkins", "Alan Taylor", "31 Oct 2013", "112 mins", "When Jane Foster is targeted by the denizens of the dark world of Svartalfheim, Thor sets out on a quest to protect her at all costs.");
+		movieDAO.createMovie(movie6);
+		
+		movies.add(movie1);
+		movies.add(movie2);
+		movies.add(movie3);
+		movies.add(movie4);
+		movies.add(movie5);
+		movies.add(movie6);
+	}
+	
+	public void createMovieGoers(){
+		int yr,mth,day;
+		Calendar calendar = Calendar.getInstance();
+		
+		day = 1;
+		mth = 2;
+		yr = 1988;
+		calendar.set(yr,mth,day,00,00,00);
+		Date date = calendar.getTime();
+		MovieGoer movieGoer1 = new MovieGoer("John", "98712354", "johnny_boy@hotmail.com",date);
+		movieGoerDAO.createMovieGoer(movieGoer1);
+		
+		day = 2;
+		mth = 2;
+		yr = 1990;
+		calendar.set(yr,mth,day,00,00,00);
+		date = calendar.getTime();
+		MovieGoer movieGoer2 = new MovieGoer("Jane", "99711234", "jane_girl@hotmail.com",date);
+		movieGoerDAO.createMovieGoer(movieGoer2);
+		
+		day = 3;
+		mth = 3;
+		yr = 1991;
+		calendar.set(yr,mth,day,00,00,00);
+		date = calendar.getTime();
+		MovieGoer movieGoer3 = new MovieGoer("Max", "96547776", "max_hot@hotmail.com",date);
+		movieGoerDAO.createMovieGoer(movieGoer3);
+		
+		day = 4;
+		mth = 5;
+		yr = 1996;
+		calendar.set(yr,mth,day,00,00,00);
+		date = calendar.getTime();
+		MovieGoer movieGoer4 = new MovieGoer("lynners", "98234554", "Awhat_miracle@hotmail.com",date);
+		movieGoerDAO.createMovieGoer(movieGoer4);
+		
+		day = 20;
+		mth = 5;
+		yr = 1992;
+		calendar.set(yr,mth,day,00,00,00);
+		date = calendar.getTime();
+		MovieGoer movieGoer5 = new MovieGoer("linus", "98234554", "whatsap-_miracle@hotmail.com",date);
+		movieGoerDAO.createMovieGoer(movieGoer5);
+		
+		day = 12;
+		mth = 12;
+		yr = 1953;
+		calendar.set(yr,mth,day,00,00,00);
+		date = calendar.getTime();
+		MovieGoer movieGoer6 = new MovieGoer("syndicate", "98234554", "boners_miracle@hotmail.com",date);
+		movieGoerDAO.createMovieGoer(movieGoer6);
+		
+	}
+	
+	public void createShowTimes(){
+		int mth,day,hr;
+		/////////////////////////////////////1st day////////////////////////////////////
+		mth = 11;
+		day= 24;
+		hr = 8;
+		Calendar calendar = Calendar.getInstance();
+		Date date = null;
+		
+		
+		Cineplex cineplex = cineplexDAO.getCineplexes().get(0);//1st cineplex
+//		mth = 11;
+//		day=13;
+//		hr = 8;
+		
+		Cinema cinema = cineplex.getCinemas().get(0);//for 1st cinema
+		for(hr=8;hr<22;hr+=3){
+			calendar.set(2013,mth,day,hr,00,00);
+			date = calendar.getTime();
+			ShowTime showTime = new ShowTime(cinema, movies.get(0), date);
+			showTimeDAO.createShowTime(showTime);
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 217:222, 225:230, 236:241, 244:249, 254:259, 262:267, 272:277, 288:293, 296:301, 307:312, 315:320, 325:330, 333:338, 343:348.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.